pkg/scheduler/predicates/ha.go

```diff
@@ -157,8 +162,19 @@ func (h *ha) Filter(instanceName string, pod *apiv1.Pod, nodes []apiv1.Node) ([]
 		scheduledNodes = append(scheduledNodes, scheduledNode)
 	}
 
+	actualReplicasNumber := 0
 	for _, pod := range podList.Items {
 		pName := pod.GetName()
+		if podInDeleteSlots(&pod, deleteSlots) {
+			klog.Infof("pod %s is contained in deleteSlots %v, will be deleted, do not count topology", pName, deleteSlots.List())
+			continue
+		}
+		actualReplicasNumber++
+		if actualReplicasNumber > int(replicas) {
+			klog.Infof("pod %s is out of range of desired replicas(%d), will be deleted, do not count topology", pName, replicas)
+			continue
+		}
```

Comment:
Is this block the similar logic of the function `isPodDesired` in `pkg/manager/member/tikv_failover.go`?
It's not correct to just check `if actualReplicasNumber > int(replicas)` to ignore Pods because the Pods may be desired and should not be ignored.